teos/src/config.rs

```diff
@@ -242,6 +242,94 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// Gets a collection of custom options using vector.
+    pub fn get_non_default_options(&self) -> Vec<(String, String)> {
```

Comment:
I don't think we will need the returned vector of string pairs for something else.
Why not log inside this method and call it `log_non_default_options` instead?

Reply:
> I don't think we will need the returned vector of string pairs for something else. Why not log inside this method and call it `log_non_default_options` instead?

yes it makes more sense.